pkg/providers/az-client.go

```diff
@@ -43,3 +46,36 @@ func (g *GraphServiceClient) GetApplicationObjectId(ctx context.Context, appId s
 	}
 	return *appObjectId, nil
 }
+
+type RoleAssignmentClient struct {
+	Client *armauthorization.RoleAssignmentsClient
+}
+
+type RoleAssignClient interface {
+	CreateRoleAssignment(ctx context.Context, objectId, roleId, scope, raUid string) error
+}
+
+var _ RoleAssignClient = &RoleAssignmentClient{}
+
+func (r *RoleAssignmentClient) CreateRoleAssignment(ctx context.Context, objectId, roleId, scope, raUid string) error {
+	log.Debug("Assigning contributor role to service principal...", "objectId", objectId, "role assignment UID", raUid, "scope", scope)
+
+	fullAssignmentId := fmt.Sprintf("/%s/providers/Microsoft.Authorization/roleAssignments/%s", scope, raUid)
+	fullDefinitionId := fmt.Sprintf("/providers/Microsoft.Authorization/roleDefinitions/%s", roleId)
+
+	params := armauthorization.RoleAssignmentCreateParameters{
+		Properties: &armauthorization.RoleAssignmentProperties{
+			PrincipalID:      &objectId,
+			RoleDefinitionID: &fullDefinitionId,
+		},
+	}
+
+	resp, err := r.Client.CreateByID(ctx, fullAssignmentId, params, nil)
```

Comment:
Here is good example of CreateByID from sdk documents - https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/authorization/armauthorization/v2#example-RoleAssignmentsClient.CreateByID

basically raUid is role assignment unique name? right?

Reply:
yeah correct, its role assignment unique ID. I will be using UUID to serve as raUid.
Thanks for the example link.

Reply:
In the SDK function, the PrincipalID property explicitly specifies that the role assignment is for a service principal, achieving the same effect as the --assignee-principal-type flag in the CLI function. Therefore, you don't need to include --assignee-principal-type in the SDK function.